docs/source/release_notes.rst

```diff
@@ -4,13 +4,29 @@ Release Notes
 -------------
 Future Release
 ==============
+    .. warning::
+        This Woodwork release uses a weak reference for maintaining a reference from the
+        accessor to the DataFrame. Because of this, chaining a Woodwork call onto a DataFrame
+        or Series initialization call can be problematic.
+
+        Instead of calling ``pd.DataFrame({'id':[1, 2, 3]}).ww.init()``, first store the DataFrame in a new
```

Comment:
Might also be worth adding something like `sample_df.copy().ww.init()` as another example here and in the start notebook below of something that will no longer work since it wasn't immediately clear to me that that was included as a part of this definition. (I'm surprised it's not something we had anywhere!)

Reply:
I don't think this (or the other example) are really typical patterns that would be followed with the current implementation, so I think the risk of users encountering this are small. 

Normally when you do something like initialize a dataframe or call copy you are going to be assigning the returned object to a variable. If we did something like `ww_df = sample_df.copy().ww.init()` the value in `ww_df` would be `None`, even with a strong reference because we don't return anything from the `.ww.init()` call. I think that is why we haven't used the pattern shown above in our code or tests - we really only used it inside a context manager where we wanted to check that errors were raised and never cared about the return value.

Anyway, all that is to say I'm not sure if adding more examples will help or just create more confusion as people read this...

Maybe I should update the text to say something like `chaining a Woodwork call onto another call that creates a new DataFrame or Series object can be problematic` instead of specifically referencing initialization?

Reply:
Updated the wording.

Reply:
looks good!